Caboodle/Compass/Compass.android.cs

```diff
@@ -13,10 +11,13 @@ public static partial class Compass
             Platform.SensorManager?.GetDefaultSensor(SensorType.Accelerometer) != null &&
             Platform.SensorManager?.GetDefaultSensor(SensorType.MagneticField) != null;
 
-        internal static void PlatformStart(SensorSpeed sensorSpeed, Action<CompassData> handler)
+        static SensorListener listener;
+        static Sensor magnetometer;
+        static Sensor accelerometer;
```

Comment:
Can we dispose/null these out when they are no longer needed?

Reply:
so they are shared system wide and if you dispose them out then they will effect other listeners from my testing :(

Reply:
Well actually.... maybe if we only allow one at a time.... let me see here. Will need to see if you can do both compass and others at the same time.

Reply:
I guess it could get ugly, but could we track/count the # of listeners we have and dispose if 0?

Reply:
What if we use a weak reference? This way we leave it up the Androids?

Reply:
#resolved